test/unit/Roave/InfectionStaticAnalysisTest/CliUtilityTest.php

```diff
@@ -0,0 +1,150 @@
+     */
+    public function testExtractArgument(
+        array $expectedNewArguments,
+        ?string $expectedArgumentValue,
+        array $arguments,
+        string $argument
+    ): void {
+        $result = CliUtility::extractArgument($arguments, $argument);
+
+        self::assertSame($expectedNewArguments, $result[0]);
+        self::assertSame($expectedArgumentValue, $result[1]);
+    }
+
+    /**
+     * @return list<array{0: (non-empty-string|null), 1: list<non-empty-string>, 2: non-empty-string}>
+     */
+    public function provideExtractionData(): array
+    {
+        return [
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config', 'configuration/psalm.xml'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config', '"configuration/psalm.xml"'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config="configuration/psalm.xml"'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config=configuration/psalm.xml"'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config="configuration/psalm.xml'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config=configuration/psalm.xml'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config=foo'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config=configuration/psalm.xml', '--psalm-config=foo'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                null,
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config'],
+                'psalm-config',
+            ],
```

Comment:
should we throw/exit on this? ( missing argument value )

current behavior is to return null as the value and remove `--psalm-config` from the argument list.

1. keep as is, another edge case.
2. throw a proper exception when `--psalm-config` is provided without a value
3. exit('--psalm-config argument value is missing');

Reply:
IMO throwing is fine.